ddtrace/contrib/aiohttp/patch.py

```diff
@@ -1,25 +1,130 @@
+from ddtrace.internal.utils.formats import asbool
+from ddtrace.vendor import wrapt
 
+from ...ext import SpanTypes
+from ...pin import Pin
+from ...propagation.http import HTTPPropagator
+from ..async_trace_utils import with_traced_module
+from ..trace_utils import ext_service
+from ..trace_utils import set_http_meta
+from ..trace_utils import unwrap
+from ..trace_utils import with_traced_module as with_traced_module_sync
+from ..trace_utils import wrap
 
-log = get_logger(__name__)
 
+log = get_logger(__name__)
 
 config._add(
-    "aiohttp",
+    "aiohttp_client",
     dict(
-        distributed_tracing=True,
+        distributed_tracing_enabled=asbool(os.getenv("DD_AIOHTTP_CLIENT_DISTRIBUTED_TRACING", False)),
     ),
 )
 
 
+class _WrappedConnectorClass(wrapt.ObjectProxy):
+    def __init__(self, obj, pin):
+        super().__init__(obj)
+        pin.onto(self)
+
+    async def connect(self, req, *args, **kwargs):
+        pin = Pin.get_from(self)
+        with pin.tracer.trace("%s.connect" % self.__class__.__name__):
+            result = await self.__wrapped__.connect(req, *args, **kwargs)
+            return result
+
+    async def _create_connection(self, req, *args, **kwargs):
+        pin = Pin.get_from(self)
+        with pin.tracer.trace("%s._create_connection" % self.__class__.__name__):
+            result = await self.__wrapped__._create_connection(req, *args, **kwargs)
+            return result
+
+
+@with_traced_module
+async def _traced_clientsession_request(aiohttp, pin, func, instance, args, kwargs):
+    method: str = get_argument_value(args, kwargs, 0, "method")
+    url = URL(get_argument_value(args, kwargs, 1, "url"))  # url can be a str or yarl.URL
+    headers = kwargs.get("headers") or {}
+
+    # Create a new span and attach to this instance (so we can
+    # retrieve/update/close later on the response)
+    # Note that we aren't tracing redirects
+    with pin.tracer.trace(
+        "aiohttp.request", span_type=SpanTypes.HTTP, service=ext_service(pin, config.aiohttp)
+    ) as span:
+        if pin._config["distributed_tracing_enabled"]:
+            HTTPPropagator.inject(span.context, headers)
+            kwargs["headers"] = headers
+        set_http_meta(
+            span,
+            config.aiohttp_client,
+            method=method,
+            url=str(url),
+            query=url.query,
+            request_headers=headers,
+        )
+
+        resp = await func(*args, **kwargs)  # ClientResponse
+
+        set_http_meta(
+            span, config.aiohttp_client, response_headers=resp.headers, status_code=resp.status, status_msg=resp.reason
+        )
+        return resp
+
+
+@with_traced_module_sync
+def _traced_clientsession_init(aiohttp, pin, func, instance, args, kwargs):
+    func(*args, **kwargs)
+    instance._connector = _WrappedConnectorClass(instance._connector, pin)
+
+
+def _patch(aiohttp):
+    Pin().onto(aiohttp)
+    pin = Pin(_config=config.aiohttp_client.copy())
```

Comment:
do we do `.copy()` on others? 🤔 it makes sense, but if you make a global change it won't apply to anything using this pin?

Reply:
Not sure how consistent we've been. Copy makes sense, else we'd have to have logic to discern whether a value was user-set or not.

> but if you make a global change it won't apply to anything using this pin

This is already the case for most/all of our global configuration (that it can't change at runtime).